File: nan21.dnet.module.tx.ui.extjs/src/main/resources/webapp/js/ds/StockComponent_Ds.js
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
	
Ext.define(Dnet.ns.tx + "StockComponent_Ds", {
	extend: 'Ext.data.Model',
	
	fields: [
		{name:"stockId", type:"string"},
		{name:"quantity", type:"float", useNull:true},
		{name:"priceLoc", type:"float", useNull:true},
		{name:"priceRef", type:"float", useNull:true},
		{name:"valueLoc", type:"float", useNull:true},
		{name:"valueRef", type:"float", useNull:true},
		{name:"id", type:"string"},
		{name:"clientId", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"notes", type:"string"},
		{name:"active", type:"boolean"},
		{name:"version", type:"int", useNull:true},
		{name:"refid", type:"string"},
		{name:"entityAlias", type:"string"},
		{name:"entityFqn", type:"string"}
	]
});

File: nan21.dnet.module.tx.presenter/src/main/java/net/nan21/dnet/module/tx/presenter/impl/inventory/model/StockComponent_Ds.java
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.tx.presenter.impl.inventory.model;

import java.math.BigDecimal;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.presenter.model.AbstractAuditableDs;
import net.nan21.dnet.module.tx.domain.impl.inventory.StockComponent;

@Ds(entity = StockComponent.class)
public class StockComponent_Ds extends AbstractAuditableDs<StockComponent> {

	public static final String f_stockId = "stockId";
	public static final String f_quantity = "quantity";
	public static final String f_priceLoc = "priceLoc";
	public static final String f_priceRef = "priceRef";
	public static final String f_valueLoc = "valueLoc";
	public static final String f_valueRef = "valueRef";

	@DsField(join = "left", path = "stock.id")
	private String stockId;

	@DsField
	private BigDecimal quantity;

	@DsField
	private BigDecimal priceLoc;

	@DsField
	private BigDecimal priceRef;

	@DsField
	private BigDecimal valueLoc;

	@DsField
	private BigDecimal valueRef;

	public StockComponent_Ds() {
		super();
	}

	public StockComponent_Ds(StockComponent e) {
		super(e);
	}

	public String getStockId() {
		return this.stockId;
	}

	public void setStockId(String stockId) {
		this.stockId = stockId;
	}

	public BigDecimal getQuantity() {
		return this.quantity;
	}

	public void setQuantity(BigDecimal quantity) {
		this.quantity = quantity;
	}

	public BigDecimal getPriceLoc() {
		return this.priceLoc;
	}

	public void setPriceLoc(BigDecimal priceLoc) {
		this.priceLoc = priceLoc;
	}

	public BigDecimal getPriceRef() {
		return this.priceRef;
	}

	public void setPriceRef(BigDecimal priceRef) {
		this.priceRef = priceRef;
	}

	public BigDecimal getValueLoc() {
		return this.valueLoc;
	}

	public void setValueLoc(BigDecimal valueLoc) {
		this.valueLoc = valueLoc;
	}

	public BigDecimal getValueRef() {
		return this.valueRef;
	}

	public void setValueRef(BigDecimal valueRef) {
		this.valueRef = valueRef;
	}
}
